Core.Collectors/Web/RateLimiter.cs

```diff
@@ -185,19 +183,37 @@ public static string GetRateLimitResource(HttpResponseHeaders responseHeaders)
 
         public async Task WaitIfNeededAsync(IAuthentication authentication, string resource = null)
         {
-            TimeSpan elapsedSinceLastLookup = this.DateTimeSystem.UtcNow - this.cacheDateUtc;
-            if (this.cachedResult == null || elapsedSinceLastLookup >= this.cacheInvalidationFrequency)
+            RateLimitTableEntity record = null;
+
+            string rowKey = RateLimitTableEntity.GetRowKey(this.OrganizationId, resource);
+            bool lookup = false;
```

Comment:
Might be slightly easier to read if you default to `true` and then invert the condition in line 195 and just set to `false` there. Saves you the else branch of the condition (lines 200-203).

Reply:
Good suggestion, thanks.